src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -72,9 +74,39 @@ public static class DependencyGraphRestoreUtility
             ILogger log,
             CancellationToken token)
         {
-            // TODO: This will flow from UI once we enable UI option to trigger reevaluation
-            var restoreForceEvaluate = false;
+            return RestoreAsync(
+                dgSpec,
+                context,
+                providerCache,
+                cacheContextModifier,
+                sources,
+                parentId,
+                forceRestore,
+                isRestoreOriginalAction,
+                additionalMessages,
+                progressReporter: null,
+                log,
+                token
+                );
+        }
 
+        /// <summary>
+        /// Restore a solution and cache the dg spec to context.
+        /// </summary>
+        public static async Task<IReadOnlyList<RestoreSummary>> RestoreAsync(
+            DependencyGraphSpec dgSpec,
+            DependencyGraphCacheContext context,
+            RestoreCommandProvidersCache providerCache,
+            Action<SourceCacheContext> cacheContextModifier,
+            IEnumerable<SourceRepository> sources,
+            Guid parentId,
+            bool forceRestore,
+            bool isRestoreOriginalAction,
+            IReadOnlyList<IAssetsLogMessage> additionalMessages,
+            IRestoreProgressReporter progressReporter,
+            ILogger log,
+            CancellationToken token)
```

Comment:
At some point, should we consider making a request/args class, similar to how `NuGet.Commands` has a `RestoreArgs` class, so we no longer have to keep adding overloads when we add new features?

Reply:
That's not a bad idea. 
I'd have a hard time deciding what belongs there. 

Everything? Only some APIs?

Reply:
not the cancellation token, but I think all the other parameters would belong in such a RestoreArgs class.

Reply:
Do you care if I do this in a future PR (right after this gets merged?)

I basically want to avoid others having to rereview this and this PR is already on the larger side.

Reply:
sure, go for it.